docs/en/engines/database-engines/overlay.md

```diff
@@ -0,0 +1,167 @@
+| `RENAME TABLE dboverlay.*` | **Rejected** — `BAD_ARGUMENTS`.                                                           |
+| `DROP TABLE dboverlay.*`   | **No-op** — ignored (allows `DROP DATABASE dboverlay` to succeed).                        |
+| `INSERT INTO dboverlay.*`  | **Pass-through** — executes against the table in the corresponding underlying DB.         |
+
+> Rationale: the facade is a **view**. Data-definition & data-mutation happen in the member databases.
+
+### Database life cycle (facade mode) {#life-cycle}
+
+* `DROP DATABASE dboverlay` — **succeeds** and **does not** cascade to members.
+
+  * Implementation details that make this safe:
+
+    * `getUUID()` returns **Nil** (no DB UUID mapping).
+    * `getTablesIterator(ctx, filter, /*skip_not_loaded=*/true)` returns **empty** (so the dropper doesn’t see tables).
+    * `empty()` returns **true** (facade reports empty during destructive passes).
+    * `dropTable()` is a **no-op** (dropper won’t fail mid-way).
+    * `drop()` is a **no-op** (doesn’t touch member DBs).
+* `isReadOnly()` returns **false** (so the server doesn’t block `DROP DATABASE` up front). Mutating table ops are still guarded individually.
+
+---
+
+## Resilience & safety {#safty}
+
+* **Null-safety**: all loops over `databases` skip `nullptr` entries; `registerNextDatabase()` rejects `nullptr`.
+  This prevents crashes from bad metadata or attach order.
+* **No-op on startup tasks** in facade mode: `loadStoredObjects`, `beforeLoadingMetadata`, `loadTablesMetadata`, `loadTableFromMetadata*`, `startup*`, `wait*`, `stopLoading` — all return immediately.
+  This avoids double registrations and keeps the overlay a pure view.
+
+---
+
+## Error codes and messages (facade mode) {#error-codes}
+
+| Scenario                                   | Error                                                                                                                                              |
+| :----------------------------------------- | :------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Overlay CREATE/ATTACH/ALTER/RENAME TABLE   | `BAD_ARGUMENTS` — “Database `<name>` is an Overlay facade (read-only view). Run this operation in an underlying database (e.g. `<first_member>`).” |
+| Overlay references itself                  | `BAD_ARGUMENTS`                                                                                                                                    |
+| Overlay references missing DB              | `BAD_ARGUMENTS`                                                                                                                                    |
+| DROP DATABASE overlay while tables “exist” | Succeeds (iterator/empty() semantics ensure no `DATABASE_NOT_EMPTY`)                                                                               |
+
+---
+
+## Logging {#logging}
```

Comment:
Let's remove this paragraph, it can be changed in future and likely someone may forget to update. It is OK to leave a sentence about `DROP` do not drop any data, and do only logging (or smth like this)